src/Aspire.Cli/Commands/UpdateCommand.cs

```diff
@@ -63,4 +126,257 @@ protected override async Task<int> ExecuteAsync(ParseResult parseResult, Cancell
+                }
+            }
+            catch
+            {
+                // If anything goes wrong, restore the backup
+                _logger.LogWarning("Update failed, restoring backup");
+                if (File.Exists(backupPath))
+                {
+                    if (File.Exists(targetExePath))
+                    {
+                        File.Delete(targetExePath);
+                    }
+                    File.Move(backupPath, targetExePath);
+                }
+                throw;
+            }
+        }
+        finally
+        {
+            // Clean up temp directories
+            CleanupDirectory(tempExtractDir);
+            CleanupDirectory(Path.GetDirectoryName(archivePath)!);
+        }
+    }
+
+    private static bool IsInPath(string directory)
+    {
+        var pathEnv = Environment.GetEnvironmentVariable("PATH");
+        if (string.IsNullOrEmpty(pathEnv))
+        {
+            return false;
+        }
+
+        var pathSeparator = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ';' : ':';
+        var paths = pathEnv.Split(pathSeparator, StringSplitOptions.RemoveEmptyEntries);
+        
+        return paths.Any(p => 
+            string.Equals(Path.GetFullPath(p.Trim()), Path.GetFullPath(directory), 
+                RuntimeInformation.IsOSPlatform(OSPlatform.Windows) 
+                    ? StringComparison.OrdinalIgnoreCase 
+                    : StringComparison.Ordinal));
+    }
+
+    private static async Task ExtractArchiveAsync(string archivePath, string destinationPath, CancellationToken cancellationToken)
+    {
+        if (archivePath.EndsWith(".zip", StringComparison.OrdinalIgnoreCase))
+        {
+            ZipFile.ExtractToDirectory(archivePath, destinationPath, overwriteFiles: true);
+        }
+        else if (archivePath.EndsWith(".tar.gz", StringComparison.OrdinalIgnoreCase))
+        {
+            await using var fileStream = new FileStream(archivePath, FileMode.Open, FileAccess.Read);
+            await using var gzipStream = new GZipStream(fileStream, CompressionMode.Decompress);
+            await TarFile.ExtractToDirectoryAsync(gzipStream, destinationPath, overwriteFiles: true, cancellationToken);
+        }
+        else
+        {
+            throw new NotSupportedException($"Unsupported archive format: {archivePath}");
+        }
+    }
+
+    private static void SetExecutablePermission(string filePath)
+    {
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            try
+            {
+                var mode = File.GetUnixFileMode(filePath);
+                mode |= UnixFileMode.UserExecute | UnixFileMode.GroupExecute | UnixFileMode.OtherExecute;
+                File.SetUnixFileMode(filePath, mode);
+            }
+            catch
+            {
+                // Best effort, ignore failures
+            }
```

Comment:
IMO this error should be at least logged, if the update is not entirely rolled back.